### PR TITLE
BottomNavMenu and page navigation

### PR DESCRIPTION
**Pull request by C4s4r**
This PR adds the basic build of the bottom navigation menu that is highly customizable and enables navigating between the different pages.
Each of these pages has its own `Navigator`, which means it will not be rebuild from ground when re-visited and can save it's state (like opened detail pages).

Furthermore this PR also adds first UI elements for further use in the future, like a Button, FlatButton and a TextField. They're all implemented in a way to enforce the usage of the CampusApp design language.

### CODE REVIEW

File: lib/pages/splash/splash_page.dart [hunk 1]
```diff
@@ -1,12 +1,151 @@
...
+  Stopwatch loadingTimer = Stopwatch();
+  Stopwatch idleTimer = Stopwatch();
+  String _directoryPath = '';
+
+  Settings? loadedSettings;
+
+  void startApp(BuildContext context) {
+    if (loadedSettings != null) {
+      // Normal app start
+      print('Initiate normal app start.');
```
Comment (domai-tb) [inline lib/pages/splash/splash_page.dart:32]: We should avoid print-Statements in production code. For debugging we could use `debugPrint()` instead, which provide some more (useful) information. Lets check out: [https://medium.com/@abhishekbedi/print-vs-debugprint-eac33a774d5a](https://medium.com/@abhishekbedi/print-vs-debugprint-eac33a774d5a)

File: pubspec.yaml [hunk 2]
```diff
@@ -27,7 +28,6 @@ dependencies:
   intl: ^0.17.0
   path_provider: ^2.0.11
   pdfx: ^2.0.1+2
-  xml: ^5.3.1
```
Comment (domai-tb) [inline pubspec.yaml:30]: We need the xml package for parsing the RUB news items. The server provides the content just as XML.

**Review by domai-tb — APPROVED** [review]
There is a critical removal in `pubspec.yaml`- the xml package. We need that to parse the RUB news content. Without that package we have to rewrite the rubnews infrastructure layer.  

I also think we should not use `print()` calls on our master branch and follow the recommendation of the [official dart linter](https://dart-lang.github.io/linter/lints/avoid_print.html).

File: pubspec.yaml (hunk 2 above)
Comment (C4s4r) [reply pubspec.yaml:30]: This was incorrectly removed, thanks for mentioning it. I have added the package again.

File: lib/pages/splash/splash_page.dart (hunk 1 above)
Comment (C4s4r) [reply lib/pages/splash/splash_page.dart:32]: Every `print()` should now be replaced with `debugPrint()`.

**Pull request merged**